contracts/interchain-token-service/InterchainTokenService.sol

```diff
@@ -678,11 +676,11 @@ contract InterchainTokenService is
         address tokenAddress = getStandardizedTokenAddress(tokenId);
         address tokenManagerAddress = getTokenManagerAddress(tokenId);
         address distributor = distributorBytes.length > 0 ? distributorBytes.toAddress() : tokenManagerAddress;
```

Comment:
token manager is always a distributor now, so don't need this?

Reply:
It allows users to specify an 'empty' distributor to make sure that no one (other than the tokenManager) can mint/burn, regardless of the chain architecture. I could remove it but I do like this potential feature.

Reply:
hmm, if the bytes are empty, why not just set `distributor` to `0` instead of the token manager. Token manager will already have mint/burn permission separately, so the distributor doesn't need to be set?

I guess a benefit of this is to allow the token manager to change the distributor in the future? And if the user doesn't want to allow that, they can explicitly pass distributor as `address(0)`. Ok, this makes more sense to me now